### ISSUE TITLE
Storefronts w/o category

### ISSUE THREAD

**Issue by kangoala** [particl/particl-desktop#1795]
Creating a storefront can be done without specifying a category. If you later try to add a listing to the storefront, you can not publish the listing, since the category selection box is empty. 

Either each storefront should be created with at least one category, or when trying to publish a listing on a storefront with no category, a message box should appear that the user has to define categories for the storefront first.

**Comment by zaSmilingIdiot** [comment]
After creating a storefront, the system prompts the user for adding categories. In so doing there is attention brought to the fact that categories for the storefront are necessary to some extent.

Setting a default category is not really a solution to those quickly clicking through the screens: it seems helpful when testing and trying to quickly create listings, but it does not lend itself or encourage quality storefront markets.  In fact, it may lead to the asking of the wrong type of questions, such as "why is there only one category available in my market". And, assuming that the user's don't know to add categories, it would largely appear as though the system forces only one category, which seems problematic and potentially leads to abandonment of the system. I'd rather that be a more focused question along the lines of "how do I create/edit/update categories for my market". And for those that don't know to add categories, it at least suggests that there is a way to put categories in, which either leads to a question of where, or self exploration.

There's possibilities of directing user's attention to the category creation, but at the same time I don't want to be forcing the addition of categories immediately after creating a storefront, before they can continue with the rest of the system. And setting default categories means additional effort to remove the defaults for those aware of modifying categories, and at the same time potentially discourages those that dont know by making it appear as a forced option.

**Comment by zaSmilingIdiot** [comment]
In saying the above, adding a prompt to the publish step can probably be looked into at some point.

**Comment by kangoala** [comment]
Yes, a default category is probably not a good idea, but a message box on the publishing step would already be helpfull I guess.
